metaflow/plugins/kubernetes/kubernetes_decorator.py

```diff
@@ -273,8 +273,10 @@ def step_init(self, flow, graph, step, decos, environment, flow_datastore, logge
             if isinstance(deco, ResourcesDecorator):
                 for k, v in deco.attributes.items():
                     # If GPU count is specified, explicitly set it in self.attributes.
-                    if k == "gpu" and v != None:
+                    if k == "gpu" and v:
                         self.attributes["gpu"] = v
+                    else:
+                        self.attributes["gpu"] = None
```

Comment:
Realizing this works locally by coincidence of `gpu` being the last attribute (late night) - will fix.